### PR TITLE
Feat/maybe interactor

### PR DESCRIPTION
**Pull request by okalman**

### CODE REVIEW

File: app/src/main/java/com/thefuntasty/mvvmsample/domain/StateInteractor.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    private var emitSuccess: Boolean = false
+
+    fun init(emitSuccess: Boolean) = apply {
+        this.emitSuccess = emitSuccess
+    }
+
+    override fun prepare(): Maybe<Boolean> = if (emitSuccess) {
+        Maybe.fromCallable {
+            true
+        }.delay(5, TimeUnit.SECONDS)
```
Comment (thefuntasty-ops) [inline app/src/main/java/com/thefuntasty/mvvmsample/domain/StateInteractor.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/thefuntasty/mvvm-android/blob/9c52775cefdb4bc11261092d9405e57c98a63a05/app/src/main/java/com/thefuntasty/mvvmsample/domain/StateInteractor.kt#L19"></span>This expression contains a magic number. Consider defining it to a well named constant.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/thefuntasty/mvvmsample/domain/StateInteractor.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    } else {
+        Maybe.empty<Boolean>().delay(5, TimeUnit.SECONDS)
```
Comment (thefuntasty-ops) [inline app/src/main/java/com/thefuntasty/mvvmsample/domain/StateInteractor.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/thefuntasty/mvvm-android/blob/9c52775cefdb4bc11261092d9405e57c98a63a05/app/src/main/java/com/thefuntasty/mvvmsample/domain/StateInteractor.kt#L21"></span>This expression contains a magic number. Consider defining it to a well named constant.</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/thefuntasty/mvvmsample/ui/login/fragment/LoginViewModel.kt [hunk 3]
```diff
@@ -1,14 +1,25 @@
...
-    private val loginInteractor: LoginInteractor
+    private val loginInteractor: LoginInteractor,
+    private val stateInteractor: StateInteractor
 ) : BaseRxViewModel<LoginViewState>() {
     override val viewState = LoginViewState()
 
+    override fun onStart() {
+        stateInteractor.init(true).execute({
+            viewState.showHeader.value = View.VISIBLE
+        },{
```
Comment (thefuntasty-ops) [inline app/src/main/java/com/thefuntasty/mvvmsample/ui/login/fragment/LoginViewModel.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/thefuntasty/mvvm-android/blob/9c52775cefdb4bc11261092d9405e57c98a63a05/app/src/main/java/com/thefuntasty/mvvmsample/ui/login/fragment/LoginViewModel.kt#L18"></span>Missing spacing after “,”</td>
    </tr>
  </tbody>
</table>

Comment (thefuntasty-ops) [inline app/src/main/java/com/thefuntasty/mvvmsample/ui/login/fragment/LoginViewModel.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/thefuntasty/mvvm-android/blob/9c52775cefdb4bc11261092d9405e57c98a63a05/app/src/main/java/com/thefuntasty/mvvmsample/ui/login/fragment/LoginViewModel.kt#L18"></span>Missing spacing before “{“</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/thefuntasty/mvvmsample/ui/login/fragment/LoginViewModel.kt [hunk 4]
```diff
@@ -17,4 +28,6 @@ class LoginViewModel @Inject constructor(
             sendEvent(NotifyActivityEvent(fullNameString))
         }
     }
+
```
Comment (thefuntasty-ops) [inline app/src/main/java/com/thefuntasty/mvvmsample/ui/login/fragment/LoginViewModel.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/thefuntasty/mvvm-android/blob/9c52775cefdb4bc11261092d9405e57c98a63a05/app/src/main/java/com/thefuntasty/mvvmsample/ui/login/fragment/LoginViewModel.kt#L31"></span>Unexpected blank line(s) before “}”</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/thefuntasty/mvvmsample/ui/login/fragment/LoginViewModel.kt [hunk 5]
```diff
@@ -17,4 +28,6 @@ class LoginViewModel @Inject constructor(
...
+
```
Comment (thefuntasty-ops) [inline app/src/main/java/com/thefuntasty/mvvmsample/ui/login/fragment/LoginViewModel.kt]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/thefuntasty/mvvm-android/blob/9c52775cefdb4bc11261092d9405e57c98a63a05/app/src/main/java/com/thefuntasty/mvvmsample/ui/login/fragment/LoginViewModel.kt#L32"></span>Needless blank line(s)</td>
    </tr>
  </tbody>
</table>

File: app/src/main/java/com/thefuntasty/mvvmsample/domain/StateInteractor.kt [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    companion object {
+        private const val DELAY_MS = 3000L
+    }
+
+    fun init(emitSuccess: Boolean) = apply {
+        this.emitSuccess = emitSuccess
+    }
+
+    override fun prepare(): Maybe<Boolean> = if (emitSuccess) {
+        Maybe.fromCallable {
```
Comment (skywall) [inline app/src/main/java/com/thefuntasty/mvvmsample/domain/StateInteractor.kt]: Co tady hodit něco jednoduššího třeba:
```
Maybe.create<Boolean> { emitter ->
            if (emitSuccess) {
                emitter.onSuccess(true)
            } else {
                emitter.onComplete()
            }
        }
```

**Review by skywall — APPROVED** [review]

**Pull request merged**